### PR TITLE
Add registration

### PR DESCRIPTION
**Pull request by jesusantguerrero**
- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
feature

- **What is the current behavior?** (You can also link to an open issue here)

- **What is the new behavior (if this is a feature change)?**
Add registration to auth class and update action workflow
- **Other information**:

### CODE REVIEW

File: src/lib/shared/KanvasErrorAPI.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+class KanvasErrorAPI extends Error {
+  readonly errors: KanvasError;
+  readonly code: number|string;
+
+  constructor(error: AxiosError) {
+    super()
+    this.errors = error.response.data.errors
+    this.code = error.code;
+    this.message = error.message
```
Comment (SparoHawk) [inline src/lib/shared/KanvasErrorAPI.ts]:
```suggestion
    super();
    this.errors = error.response.data.errors;
    this.code = error.code;
    this.message = error.message;
```

File: .github/workflows/node.js.yml [hunk 2]
```diff
@@ -5,25 +5,29 @@ name: Node.js CI
...
+        with:
+          node-version: ${{ matrix.node-version }}
+      - run: npm ci
+      - run: npm run build --if-present
+      - run: npm run test:unit
+    env:
+      ENDPOINT: ${{ secrets.APP_ENDPOINT }}
+      APP_KEY: ${{ secrets.APP_KEY }}
+      APP_USER: ${{ secrets.APP_USER }}
+      APP_PASS: ${{ secrets.APP_PASS }}
```
Comment (SparoHawk) [inline .github/workflows/node.js.yml:33]: Is there a way to load environment variables from the actual env files? (eg: `env.production.local`)

File: src/lib/auth/auth.spec.ts [hunk 3]
```diff
@@ -1,53 +1,97 @@
 /* eslint-disable functional/no-let */
 import test from 'ava';
+import sinon, { SinonSandbox } from 'sinon';
```
Comment (SparoHawk) [inline src/lib/auth/auth.spec.ts]:
```suggestion
```

File: src/lib/auth/auth.spec.ts [hunk 4]
```diff
@@ -1,53 +1,97 @@
...
 
 import config from '../../config';
+import KanvasErrorAPI from '../shared/KanvasErrorAPI';
```
Comment (SparoHawk) [inline src/lib/auth/auth.spec.ts:5]: I think we should have a dedicated folder for error types. Opinion?

File: src/lib/auth/auth.spec.ts [hunk 5]
```diff
@@ -1,53 +1,97 @@
...
-    appKey: config.appkey,
-  },
+  auth = new Auth(
+    {
+      endpoint: config.apps.GEWAER_API,
+      appKey: config.appkey,
+    },
     new TokenProvider({
-      domain: config.domain || null
+      domain: config.domain || null,
```
Comment (SparoHawk) [inline src/lib/auth/auth.spec.ts]:
```suggestion
      domain: config.domain || null
```

File: src/lib/auth/auth.ts [hunk 6]
```diff
@@ -2,10 +2,11 @@
 /* eslint-disable functional/prefer-readonly-type */
 /* eslint-disable functional/no-this-expression */
 /* eslint-disable functional/no-class */
-import axios, { AxiosInstance } from 'axios';
+import qs from "qs";
 import { EventEmitter } from 'tsee';
```
Comment (SparoHawk) [inline src/lib/auth/auth.ts]:
```suggestion
```

File: src/lib/auth/auth.ts [hunk 7]
```diff
@@ -2,10 +2,11 @@
...
 
 import ClientOptions from '../shared/clientOptions';
+import HttpClient from '../shared/httpClient';
```
Comment (SparoHawk) [inline src/lib/auth/auth.ts]:
```suggestion
```

File: src/lib/shared/httpClient.ts [hunk 8]
```diff
@@ -0,0 +1,36 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable functional/no-throw-statement */
+/* eslint-disable functional/immutable-data */
+/* eslint-disable functional/no-this-expression */
+/* eslint-disable functional/no-class */
+import axios, { AxiosInstance, AxiosRequestConfig} from "axios";
+
+import TokenProvider from "../auth/tokenProvider";
+
```
Comment (SparoHawk) [inline src/lib/shared/httpClient.ts]:
```suggestion
import TokenProvider from "../auth/tokenProvider";
```

File: src/lib/shared/httpClient.ts [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+import KanvasErrorAPI from "./KanvasErrorAPI";
+import ClientOptions from "./clientOptions";
+
+ export default class HttpClient {
+   readonly http: AxiosInstance;
+   readonly tokenProvider: TokenProvider
```
Comment (SparoHawk) [inline src/lib/shared/httpClient.ts]:
```suggestion
   readonly tokenProvider: TokenProvider;
```

File: src/lib/shared/httpClient.ts [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+
+   constructor(options: ClientOptions, tokenProvider: TokenProvider) {
+    this.http = axios.create({
+      baseURL: options.endpoint
+    });
+    this.tokenProvider = tokenProvider
```
Comment (SparoHawk) [inline src/lib/shared/httpClient.ts]:
```suggestion
    this.tokenProvider = tokenProvider;
```

**Review by SparoHawk — CHANGES_REQUESTED** [review]

File: .github/workflows/node.js.yml (hunk 2 above)
Comment (jesusantguerrero) [reply .github/workflows/node.js.yml:33]: we are using this for test in github actions. We can provide the variables in proyect -> settings -> secrets

Comment (jesusantguerrero) [reply .github/workflows/node.js.yml:33]: https://github.com/kyanvasu/client-sdk/settings/secrets/actions

File: src/lib/auth/auth.spec.ts (hunk 4 above)
Comment (jesusantguerrero) [reply src/lib/auth/auth.spec.ts:5]: I agree. it is a good option.

**Review by SparoHawk — APPROVED** [review]

**Pull request merged**